File: 19-meca_flu-eloudrhiri-jacquet-navez/pressure.m
function p = pressure(dom,s,rho)
%% CONSTANTS
[row,column] = size(dom);
p = NaN(size(dom));

g = 9.81; % m/s^2
C = 0; % arbitrary

%% COMPUTATION
s = s.^2;

for i = 1:row
    for j = 1:column
        if(dom(i,j) ~=0)
            p(i,j) = rho*g*(C - (s(i,j)/(2*g)));
        end 
    end 
end
end 

File: pressure.py
import numpy as np


def pressure(dom, s, rho):
    row = np.size(dom, 0)
    col = np.size(dom, 1)

    p = np.zeros((row, col))
    p[:] = np.nan

    g = 9.81
    c = 0
    s = np.square(s)

    for i in range(0, row - 1):
        for j in range(0, col - 1):
            if dom[i, j] != 0:
                p[i, j] = rho * g * (c - (s[i, j] / (2 * g)))
    return p
